File: clases 25-11-2025/Tipos de clases c#/clase normal con interfaz/Persona.cs
public class Persona : IPersona
{
    public string Nombre { get; set; }
    public int Edad { get; set; }

    public Persona(string nombre, int edad)
    {
        Nombre = nombre;
        Edad = edad;
    }

    // Constructor copia
    public Persona(Persona otra)
    {
        Nombre = otra.Nombre;
        Edad = otra.Edad;
    }

    public void Saludar()
    {
        Console.WriteLine($"Hola, soy {Nombre} y tengo {Edad} años.");
    }
}

File: clases 25-11-2025/tipos de clases Java/clase normal con interfaz/Persona.java
public class Persona implements IPersona {
    public String nombre;
    public int edad;

    public Persona(String nombre, int edad) {
        this.nombre = nombre;
        this.edad = edad;
    }

    // Constructor copia
    public Persona(Persona otra) {
        this.nombre = otra.nombre;
        this.edad = otra.edad;
    }

    @Override
    public void saludar() {
        System.out.println("Hola, soy " + nombre + " y tengo " + edad + " años.");
    }
}
